### PR TITLE
Initial code for placing a word and refilling the LetterRack

### PR DESCRIPTION
**Pull request by patelhrid**
Initial code for my user story. Yet to be tested, since other functionality from the rest of the program is required. Implements the facade design pattern to hide inner workings, and hopefully minimizes code smells. The controller interface is also empty for the time being, which will be updated as needed later. Also, the Player class that has been imported will need to be changed to the one with the correct package. Another note: checking the word in the dictionary also needs to be added, but this can only be done after we push Pete's dictionary code to main.

### CODE REVIEW

**Review by petesaiacct — APPROVED** [review]
This is really efficient code. Well constructed. I really wish we had communicated more clearly as it does a lot of what I have been spending hours on creating in the TurnBuilder, but thats my problem:)


Your changes to player.java are that in reference to the interface implementation that Ethan was talking about? Or is this to the Abstract?

I'm approving as I need to make a pull from this to reformat my design. And it doesn't look like it conflicts with any major structures.

**Review by ethanhonest — APPROVED** [review]
Looks like a good start to your use case. I think we're going with the design of having one controller and one presenter, so I don't think you need a controller and presenter specifically for your use case - you'll just need an input and output boundary. The presenter class will implement your outputboundary, so it will implement all of the methods (i.e. prepareFailView) that you include in your outputboundary interface

File: src/main/java/place_word_refill_user_story/PlaceWordRefillController.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package place_word_refill_user_story;
+
+public interface PlaceWordRefillController {
+    // TODO: What even goes here?
```
Comment (JosephUofT) [inline src/main/java/place_word_refill_user_story/PlaceWordRefillController.java:4]: I believe this is something we can address with whoever is writing the code for the bag class

**Review by JosephUofT — APPROVED** [review]
This looks pretty good. I checked the code for the bag and there doesn't seem to be any conflict. In fact, it appears that some of the SOLID principles are really properly implemented. Overall, I think this pull request is good to go. I do however think we should double check that this does not conflict with what Pete is working on. I remember that he also was working on some of the same things that you worked on in this pull request.

File: src/main/java/place_word_refill_user_story/PlaceWordRefillPresenter.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package place_word_refill_user_story;
+
+public interface PlaceWordRefillPresenter {
+    PlaceWordRefillResponseModel prepareSuccessView(PlaceWordRefillResponseModel placeWordRefillResponseModel);
+
+    PlaceWordRefillResponseModel prepareFailView(String error);
```
Comment (JosephUofT) [inline src/main/java/place_word_refill_user_story/PlaceWordRefillPresenter.java:6]: We still need to figure out exactly what the code for this is

**Review by JosephSun2003 — APPROVED** [review]
Seems code is good.

**Pull request merged**